### ISSUE TITLE
props is unused in script-setup

### ISSUE THREAD

**Issue by fvclaus** [vuejs/eslint-plugin-vue#1753]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 6.8.0
- **eslint-plugin-vue version:** 7.20.
- **Node version:** 14.17.3
- **Operating System:** Ubuntu 20.04.3 LTS

**Please show your full configuration:**

```json5
{
    "root": true,
    "env": {
      "node": true
    },
    "globals": {
      "defineProps": "readonly"
    },
    "rules": {
      "vue/script-setup-uses-vars": "error"
    },
    "extends": [
      "plugin:vue/vue3-essential",
      "eslint:recommended",
      "@vue/typescript",
      "@vue/standard",
      "@vue/typescript/recommended"
    ],
    "parserOptions": {
      "parser": "@typescript-eslint/parser",
      "ecmaVersion": 2020
    }
  }
```

**What did you do?**	

```vue
<script setup lang="ts">
const props = defineProps({
  msg: {
    type: String,
    defaultValue: ''
  }
})
</script>
```

**What did you expect to happen?**
Props is not marked as unused

**What actually happened?**

warning: 'props' is assigned a value but never used (@typescript-eslint/no-unused-vars) at src/components/HelloWorld.vue:33:7:

**Repository to reproduce this issue**

Project created using `vue create ...`, `vue add typescript`, `vue add eslint`.
https://github.com/fvclaus/vue-plugin-eslint-props-not-defined

I also tried the advice from here https://github.com/vuejs/eslint-plugin-vue/issues/1617#issuecomment-910169791

**Comment by thomasklein-winemaker** [comment]
+1

**Comment by ota-meshi** [comment]
Thank you for posting issue.

If you don't use the props variable, you can remove the variable declaration.

```vue
<script setup lang="ts">
defineProps({
  msg: {
    type: String,
    defaultValue: ''
  }
})
</script>
```
